packages/flame/test/components/component_test.dart

```diff
@@ -353,10 +353,33 @@ void main() {
         expect(componentD.visited, false);
         expect(componentE.visited, false);
       });
+
+      testWithFlameGame(
+        'child returns the first child on the matching type',
+        (game) async {
+          await game.ensureAdd(ComponentA());
+          await game.ensureAdd(ComponentB());
+
+          final childA = game.child<ComponentA>();
+          expect(childA, isNotNull);
+          expect(childA, isA<ComponentA>());
+
+          final childB = game.child<ComponentB>();
+          expect(childB, isNotNull);
+          expect(childB, isA<ComponentB>());
+
+          final nonExistentChild = game.child<SpriteComponent>();
+          expect(nonExistentChild, isNull);
+        },
+      );
     });
   });
 }
 
+class ComponentA extends Component {}
+
+class ComponentB extends Component {}
```

Comment:
Are these two needed? Can't you just create new components in the test directly?

Reply:
Ofc they are needed, I should stop reviewing things right after I have woken up in the morning. 😅